src/pages/core/external-plugins/app-data.md

```diff
@@ -248,3 +349,178 @@ let write_to_app_data_plugin_ix = WriteExternalPluginAdapterDataV1CpiBuilder::ne
 {% /dialect %}
 
 {% /dialect-switcher %}
+
+## Reading Data from the AppData Plugin
+
+Data can be both read on chain progams and external sources pulling account data.
+
+### Fetch the Raw Data
+
+The first step to deserializing the data stored in an AppData plugin is to fetch the raw data and check the schema field which dictates the format in which the data is stored before serialization.
+
+{% dialect-switcher title="Fetching AppData Raw Data" %}
+{% dialect title="JavaScript" id="js" %}
+
+```ts
+const assetId = publicKey('11111111111111111111111111111111')
+const dataAuthority = publicKey('33333333333333333333333333333333')
+
+const asset = await fetchAsset(umi, assetId)
+
+let appDataPlugin = asset.appDatas?.filter(
+  (appData) => (appData.authority.address = dataAuthority)
+)
+
+let data
+let schema
+
+// check if AppData plugin with the given authority exists
+if (appDataPlugin && appDataPlugin.length > 0) {
+  //save plugin data to `data`
+  data = appDataPlugin[0].data
+
+  // save plugin schema to `schema`
+  schema = appDataPlugin[0].schema
+}
+```
+
+{% /dialect %}
+
+{% dialect title="Rust" id="rust" %}
+
+```rust
+
+let plugin_authority = ctx.accounts.authority.key();
+
+let asset = BaseAssetV1::from_bytes(&data).unwrap();
```

Comment:
I wonder if for Rust if we should also show the second approach using `Asset::from_bytes()` and then using the `AppDataWithData::data_offset` and `AppDataWithData::data_len` to calculate the slice and then parse it from there?  Example of this would be in `create_with_external_plugin.rs` where it says "Second, get app data offset and length from a full `Asset` deserialization".

On the other hand maybe we don't want to showcase the full-on `Asset` and `Collection` as much.  But they are there and probably viable for any non-program Rust user (I think they work in programs now too as long as assets don't have too many plugins (After the stack optimizations it worked fine for me in the IslandMon Anchor game with a big `Attributes` plugin).